File: Бойко Дмитро/Лабораторна робота №2/Лабораторна робота №2/Container.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Лабораторна_робота__2
{
    public abstract class Container
    {
        public int ID { get; protected set; }
        public int Weight { get; protected set; }

        public abstract double Consumption();

        public bool Equals(Container other)
        {
            return this.GetType() == other.GetType() && this.ID == other.ID && this.Weight == other.Weight;
        }
    }

    // BasicContainer class
    public class BasicContainer : Container
    {
        public BasicContainer(int id, int weight)
        {
            ID = id;
            Weight = weight;
        }

        public override double Consumption()
        {
            return Weight * 2.5;
        }
    }

    // HeavyContainer class
    public class HeavyContainer : Container
    {
        public HeavyContainer(int id, int weight)
        {
            ID = id;
            Weight = weight;
        }

        public override double Consumption()
        {
            return Weight * 3.0;
        }
    }

    // RefrigeratedContainer class
    public class RefrigeratedContainer : HeavyContainer
    {
        public RefrigeratedContainer(int id, int weight) : base(id, weight) { }

        public override double Consumption()
        {
            return Weight * 5.0;
        }
    }

    // LiquidContainer class
    public class LiquidContainer : HeavyContainer
    {
        public LiquidContainer(int id, int weight) : base(id, weight) { }

        public override double Consumption()
        {
            return Weight * 4.0;
        }
    }
}

File: Гель Олег/Lab3/container.py
from abc import ABC, abstractmethod

class Container(ABC):
    def __init__(self, ID, weight):
        self.ID = ID
        self.weight = weight

    @abstractmethod
    def consumption(self):
        pass

    def equals(self, other):
        return (
            isinstance(other, Container) and
            self.ID == other.ID and
            self.weight == other.weight
        )

class BasicContainer(Container):
    def __init__(self, ID, weight):
        if weight <= 3000:
            super().__init__(ID, weight)
        else:
            raise ValueError("Weight of BasicContainer should be <= 3000.")

    def consumption(self):
        return 2.50 * self.weight

class HeavyContainer(Container):
    def __init__(self, ID, weight):
        if weight > 3000:
            super().__init__(ID, weight)
        else:
            raise ValueError("Weight of HeavyContainer should be > 3000.")

    def consumption(self):
        return 3.00 * self.weight

class RefrigeratedContainer(HeavyContainer):
    def __init__(self, ID, weight):
        super().__init__(ID, weight)

    def consumption(self):
        return 5.00 * self.weight

class LiquidContainer(HeavyContainer):
    def __init__(self, ID, weight):
        super().__init__(ID, weight)

    def consumption(self):
        return 4.00 * self.weight
